File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03574/p03574_num7_parsed.py
# read input
H, W = map(int, input().split())
grid = [input() for _ in range(H)]

# define the directions to check for adjacent squares
directions = [(0, 1), (0, -1), (1, 0), (-1, 0), (1, 1), (-1, -1), (1, -1), (-1, 1)]

# iterate over each square in the grid
for i in range(H):
    for j in range(W):
        # skip if the square contains a bomb
        if grid[i][j] == '#':
            continue
        
        # count the number of adjacent bomb squares
        count = 0
        for dx, dy in directions:
            ni, nj = i + dx, j + dy
            # check if adjacent square is within the grid
            if ni >= 0 and ni < H and nj >= 0 and nj < W:
                if grid[ni][nj] == '#':
                    count += 1
        
        # replace the empty square with the count
        grid[i] = grid[i][:j] + str(count) + grid[i][j+1:]

# print the grid
for row in grid:
    print(row)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03574/p03574_num7_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int H, W;
    cin >> H >> W;
    vector<string> grid(H);
    for (int i = 0; i < H; i++) {
        cin >> grid[i];
    }
    
    vector<vector<int>> directions = {{-1, -1}, {-1, 0}, {-1, 1}, {0, -1}, {0, 1}, {1, -1}, {1, 0}, {1, 1}};
    
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            if (grid[i][j] == '#') {
                continue;
            }
            
            int count = 0;
            for (int k = 0; k < 8; k++) {
                int nx = i + directions[k][0];
                int ny = j + directions[k][1];
                if (nx >= 0 && nx < H && ny >= 0 && ny < W && grid[nx][ny] == '#') {
                    count++;
                }
            }
            
            grid[i][j] = count + '0';
        }
    }
    
    for (int i = 0; i < H; i++) {
        cout << grid[i] << endl;
    }
    
    return 0;
}